### PR TITLE
Preparations to 1.0 release

### PR DESCRIPTION
**Pull request by nailor**
Sorry for a huge PR, try reviewing comment by comment if possible.

This PR does three major things:

1. Tests are no longer run by default
2. Default installation path is changed
3. Next version will be tagged as 1.0 (YAY!)

### CODE REVIEW

File: dh_virtualenv/cmdline.py [hunk 1]
```diff
@@ -40,19 +40,38 @@ def parse_args(self, args=None, values=None):
...
-            warnings.simplefilter('default')
-            warnings.warn('Use of --pypi-url is deprecated. Use --index-url intead',
-                          DeprecationWarning)
-    if parser.values.index_url:
-        # We've already set the index_url, which means that we have both
-        # --index-url and --pypi-url passed in.
-        raise OptionValueError('Deprecated --pypi-url and the new '
-                               '--index-url are mutually exclusive')
-    parser.values.index_url = value
+def _check_for_depreacted_options(
```
Comment (jhermann) [inline dh_virtualenv/cmdline.py]: typo. (deprecated)

**Comment by jooon** [comment]
+1

**Comment by aRkadeFR** [comment]
I take advantage of this PR to congrats you to this awesome tool :)
And the 1.0 release LGTM :)

File: dh_virtualenv/cmdline.py [hunk 2]
```diff
@@ -40,19 +40,38 @@ def parse_args(self, args=None, values=None):
...
+            # We've already set the index_url, which means that we have both
+            # --index-url and --pypi-url passed in.
+            raise OptionValueError('Deprecated --pypi-url and the new '
+                                   '--index-url are mutually exclusive')
+        parser.values.index_url = value
+    elif opt_str in ('--no-test', '--setuptools-test'):
+        if opt_str == '--no-test':
+            with warnings.catch_warnings():
+                warnings.simplefilter('default')
+                warnings.warn('Use of --no-test is deprecated and has now '
```
Comment (daenney) [inline dh_virtualenv/cmdline.py]: I think "now" should be "no" here.

File: doc/info.rst [hunk 3]
```diff
@@ -24,18 +24,37 @@ In the sequence the ``dh_virtualenv`` is inserted right after
...
-* Deprecate ``--pypi-url`` in favour of ``--index-url``
-* Support upgrading pip to the latest release with ``--upgrade-pip``
+* **Backwards incompatible** Change the default install root to
+  ``/opt/venvs``. This is due to the old installation root
+  (``/usr/share/python``) clashing with Debian provided Python
+  utilities. To maintain the old install location, use
+  :envvar:`DH_VIRTUALENV_INSTALL_ROOT` and point it to
+  ``/usr/share/python``.
+* **Backwards incompatible** By default, do not run `setup.py test`
+  upon building. The :option:`--no-test` flag has no longer any effect. To
```
Comment (daenney) [inline doc/info.rst]: "no longer has any effect"

File: doc/info.rst [hunk 4]
```diff
@@ -24,18 +24,37 @@ In the sequence the ``dh_virtualenv`` is inserted right after
...
+  get the old behaviour, use :option:`--setuptools-test` flag instead.
```
Comment (daenney) [inline doc/info.rst]: "use the :option:...."

File: doc/usage.rst [hunk 5]
```diff
@@ -17,11 +17,11 @@ To use an alternative install prefix, add a line like
...
 
 to the top of your ``debian/rules`` file. dh_virtualenv will use
-DH_VIRTUALENV_INSTALL_ROOT instead of ``/usr/share/python`` when it
-constructs the install path.
+:envvar:`DH_VIRTUALENV_INSTALL_ROOT` instead of ``/opt/venvs``
+when it constructs the install path.
 
 To use an install suffix other than the package name, call the
-``dh_virtualenv`` command using with the ``--install-suffix``
+``dh_virtualenv`` command using with the :option:`--install-suffix`
```
Comment (daenney) [inline doc/usage.rst]: The "with" here is unnecessary.

File: doc/usage.rst [hunk 6]
```diff
@@ -54,6 +54,20 @@ However, the tool makes a few assumptions of your project's structure:
 After these are place, you can just build the package with your
 favorite tool!
 
+Environmental variables
```
Comment (daenney) [inline doc/usage.rst]: It's "environment variables", not "environmental". The first one has to do with computers, the second with the physical world and the climate.

File: doc/usage.rst [hunk 7]
```diff
@@ -54,6 +54,20 @@ However, the tool makes a few assumptions of your project's structure:
...
+=======================
+
+Certain environmental variables can be used to customise the behaviour
```
Comment (daenney) [inline doc/usage.rst]: Idem here.

File: doc/usage.rst [hunk 8]
```diff
@@ -141,11 +157,13 @@ few command line options:
...
 
-.. cmdoption:: --no-test
+.. cmdoption:: --setuptools-test
+
+   .. versionadded:: 1.0
 
-   Skip running ``python setup.py test`` after dependencies and the
-   package is installed. This is useful if the Python code is packaged
-   using distutils and not setuptools.
+   Force running ``python setup.py test`` when building the package.
```
Comment (daenney) [inline doc/usage.rst]: I wouldn't say "force running" here but simply "run". Forcing suggests there's a need to apply some pressure to get this done whereas we simply flipped the default.

File: doc/usage.rst [hunk 9]
```diff
@@ -250,44 +273,62 @@ In addition the separation of build and install steps makes it
 possible to use ``debian/install`` files to include built files into
 the Debian package. This is not possible with the sequencer addition.
 
-The build system honors the ``DH_VIRTUALENV_INSTALL_ROOT`` environment
-variable. Arguments can be passed to virtualenv by setting
-``DH_VIRTUALENV_ARGUMENTS``. For example:
+The build system honors the :envvar:`DH_VIRTUALENV_INSTALL_ROOT`
+environment variable. Following other environmental variables can be
```
Comment (daenney) [inline doc/usage.rst]: The "environmental" is back.

File: test/test_cmdline.py [hunk 10]
```diff
@@ -71,8 +71,22 @@ def test_pypi_url_creates_deprecation_warning():
...
+def test_no_test_creates_deprecation_warning():
+    parser = cmdline.get_default_parser()
+    with warnings.catch_warnings(record=True) as w:
+        parser.parse_args([
+            '--no-test',
+        ])
+    eq_(len(w), 1)
+    ok_(issubclass(w[0].category, DeprecationWarning))
+    eq_(str(w[0].message),
+        'Use of --no-test is deprecated and has now effect. '
```
Comment (daenney) [inline test/test_cmdline.py]: "now" -> "no"

**Comment by nailor** [comment]
Thanks @daenney for proofreading! :)

File: README.md [hunk 11]
```diff
@@ -41,7 +41,7 @@ NOTE: Be aware that the configuration in debian/rules expects tabs instead of sp
 
 Once the package is built, you have a virtualenv contained in a Debian
 package and upon installation it gets placed, by default, under
-`/usr/share/python/<packagename>`.
+`/opt/venvs/<packagename>`.
```
Comment (chriskuehl) [inline README.md]: I really don't think `/opt` is an appropriate place for Debian packages to be installing applications by default.

I left a more detailed comment: https://github.com/spotify/dh-virtualenv/issues/159#issuecomment-239040540

Comment (daenney) [inline README.md]: The location is configurable so you can change it for yourself.

By the very nature of this tooling it violated the Debain packaging principles as its a "bring the whole environment" form of packaging. If you look at every other package that comes in this form, be it the Puppet AIO packaging, GitLab Omnibus and a bunch more, they all live in `/opt`.

Comment (chriskuehl) [inline README.md]: :+1: as long as it's configurable that sounds fine to me.

Comment (daenney) [inline README.md]: Take a look at the docs change that are part of this commit, specifically the things related to `DH_VIRTUALENV_INSTALL_ROOT`. That'll allow you to relocate an installation to `/usr/share` for example or anywhere else you would like 😄.

**Pull request merged**